pages/today.vue

```diff
@@ -1,11 +1,38 @@
 <script setup lang="ts">
 import { format } from 'date-fns';
+import type { Database } from '@/types/database.types';
+const user = useSupabaseUser();
+const supabase = useSupabaseClient<Database>();
 
 const date = Date.now();
+const user_id = user.value!.id;
+
+const monthDay = extractMonthDay(date);
+const { data: entries } = await useAsyncData('entries', async () => {
```

Comment:
this can stay here for now, but you could also consider extracting this as `useEntries` or something like that, when you feel like you have to :)

Reply:
By this, did you mean that it might be a better idea to put the fetch request for the supabase data within a useEntries composable?

Reply:
Ah I meant to extract the whole `useAsyncData('entries', ....` as a composable.

```js
export const useEntries = (userId, monthDay) => {
  const { data: entries } = await useAsyncData('entries', async () => {
    const { data } = await supabase
      .from('entries')
      .select()
      .eq('user_id', userId)
      .eq('month_day', monthDay)
      .order('created_ts', { ascending: false });
    return data;
  });
  return entries;
}
```

Reply:
> Also, in JS, we mostly use camelCase, so it's better to use `userId` whenever possible, instead of user_id :)

Ah yes, I think I got it confused with my supabase db's user_id, thank you for catching it! 

So I initially thought you meant that I should extract the whole useAsyncData, except when I tried that initially,  it was telling me I cannot use `await` keyword because `useEntries` isn't an async function.

![CleanShot 2024-03-05 at 20 01 23](https://github.com/boyeonihn/manyaday/assets/95451902/7eb01771-6b09-4056-bfc0-ddd258fbad1f)

So I made `useEntries` an async function, then used `await` in front of the `useEntries` composable at `today.vue`


![CleanShot 2024-03-05 at 20 02 34](https://github.com/boyeonihn/manyaday/assets/95451902/ecdfe989-92ac-4695-b8b1-12fdb52486b9)
![CleanShot 2024-03-05 at 20 03 08](https://github.com/boyeonihn/manyaday/assets/95451902/5dd8d30f-68b4-4345-a5ac-38097174cdf3)

The code works as expected once I make the await/async keyword modifications -- but is this the correct way to go about it? Still pretty new to composables which is I guess why I thought I shouldn't be making composables async.

Reply:
> I thought I shouldn't be making composables async

that's the exact impression I had before.

However, you await that async composable to render a component, and it means, on the SSR, the server will make sure to fetch the data, and then it renders the page. Otherwise, it would've rendered a blank page with no entries data. Let me know if it's still unclear!

Reply:
Ah, got it!! Still many long ways to go on learning this 😆 My first Nuxt composable!!